projects/ardin_mb/state_handler.cc

```diff
@@ -367,14 +486,111 @@ bool StateHandler::checkAntPosition()
     }
 }
 //----------------------------------------------------------------------------
-void StateHandler::resetAntPosition()
+void StateHandler::loadRoute(const std::string &filename)
 {
-    if(m_Route.size() > 0) {
-        m_Pose = m_Route[0];
+    // If loading route is successful
+    m_Route.load(filename);
+
+    // Get bounds of route
+    const auto &routeMin = m_Route.getMinBound();
+    const auto &routeMax = m_Route.getMaxBound();
+
+    // Create vector field geometry to cover route bounds
+    m_VectorField.createVertices(routeMin[0] - 20_cm, routeMax[0] + 20_cm, 20_cm,
+                                 routeMin[1] - 20_cm, routeMax[1] + 20_cm, 20_cm);
```

Comment:
I'm guessing the +-20cm is just so that there's a bit of a margin in the display? Maybe a comment is needed